parsec/mca/device/device.h

```diff
@@ -283,6 +285,38 @@ parsec_device_find_function(const char* function_name,
     MCA_BASE_VERSION_2_0_0, \
         "device", 2, 0, 0
 
+/**
```

Comment:
This approach seems costly and not more difficult to implement (as an example I'm not sure it is working without marking `data_in_array_size` as volatile). I think a simple approach is to allocate the `data_in_from_device` array during the device attach. If the specialized device->attach allocate it's own array then so be it, otherwise the `parsec_mca_device_attach` will allocate it as needed.

Reply:
Right, I didn't know that attach happens after all modules have been selected, this is a better place than dynamic sizing of this array. See commit 38b7b3e16f126731f8cef41f97d91d085ae4b56c